elasticdl/python/master/servicer.py

```diff
@@ -67,7 +71,9 @@ def report_task_result(self, request, _):
             logger.warning("Worker reported error: " + request.err_message)
             self._task_d.report(request, False)
         else:
-            self._task_d.report(request, True)
+            complete_time = self._task_d.report(request, True)
+            self._task_complete_times.pop(0)
+            self._task_complete_times.append(complete_time)
```

Comment:
`self._task_complete_times` is from the last 20 completed tasks.
If training tasks take a long time, but the evaluation task can finish very quickly, and the worker gets 20 evaluation tasks in sequence. This will make the mean of `self._task_complete_times` very small, and timeout following training tasks.
Vice versa

Reply:
How about use all the historial task complete time?

Reply:
Or we need to distinguish `training` task and `eval` task.
`total_training_time`
`num_training_task`
`total_eval_time`
`num_eval_task`
```
if (num_training_task > threshold(such as 20) and
     current_training_task_running_time > total_training_time / num_training_task
):
  timeout the task
 ```

Reply:
Done